scaleway/resource_domain_record_test.go

```diff
@@ -0,0 +1,592 @@
+package scaleway
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	domain "github.com/scaleway/scaleway-sdk-go/api/domain/v2beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+var (
+	// prevent using production domain for testing
+	reservedDomains = []*regexp.Regexp{
+		regexp.MustCompile(`.*iliad.*`),
+		regexp.MustCompile(`.*\.free\..*`),
+		regexp.MustCompile(`.*\.online\..*`),
+		regexp.MustCompile(`.*scaleway.*`),
+		regexp.MustCompile(`.*dedibox.*`),
+	}
+	testDomain = ""
+)
+
+func init() {
```

Comment:
Please remove this init() and set one const dns_zone to test. Please consider that the dns should be added by our team to be consider in the nightly test.

Reply:
Another solution could be to have the domain named changed in the cassettes to example.com for instance. So that way when we record cassettes, we don't care too much about the DNS zone being used. 
When the cassettes are running they will use example.com

Reply:
I've removed the "take the first domain available" part and force using the "TF_TEST_DOMAIN" env var. But imo we should keep the init func, this is a build-in Go for test files. The init check if the var is set and it's not a production scaleway domain.
And if I'm using a const for dns_zone, a user without access to the test domain can't run the test without changing the code.
Using example.com will be the same